services/httpd/handler_test.go

```diff
@@ -2703,6 +2717,35 @@ func TestHandlerDebugVars(t *testing.T) {
 			}
 		})
 	})
+
+	t.Run("checks crypto diagnostic handling", func(t *testing.T) {
+		h := NewHandler(false)
+		// intentionally leave out "ensureFIPS" to test that code path
+		h.Monitor.DiagnosticsFn = newDiagFn(
+			map[string]*diagnostics.Diagnostics{
+				"crypto": diagnostics.RowFromMap(map[string]interface{}{
+					"FIPS":           true,
+					"passwordHash":   "pbkdf2-sha256",
+					"implementation": "BoringCrypto",
+				}),
+			})
+		req := MustNewRequest("GET", "/debug/vars", nil)
+		w := httptest.NewRecorder()
+		h.ServeHTTP(w, req)
+		got := read(t, w.Body, Ignored...)
+		exp := map[string]interface{}{
+			"crypto": map[string]interface{}{
+				"FIPS":           true,
+				"ensureFIPS":     nil,
```

Comment:
Do you want to omit this entirely, rather than having a `nil` entry?

Reply:
The `nil` is intentional. Here's my reasoning:

Assumptions:

- Seeing `nil` values means enterprise was not updated to generate the appropriate diagnostic values.
- You should not see `nil` in release versions, because enterprise and OSS agree on the appropriate attributes.
- Incomplete data is better than no data, as long as there's no misleading data.

Option 1: If enterprise doesn't provide an attribute value, `GET /debug/vars` will return a 500. This is what other sections like `build` do.
I don't like this option because OSS regression tests will pass, but enterprise currently has no tests for `/debug/vars`. Hopefully we'd catch this, but maybe not. We might have a situation where a customer can't pull `/debug/vars`, even though it has lots of other useful info.

Option 2: Leave the field out if enterprise doesn't set it.
Let's say in the future we add PAM support, and add `PAMEnabled` and `PAMModules` to OSS, but enterprise hasn't been updated yet. If we see `/debug/vars` without `PAMEnabled` and `PAMModules`, is that because the version is too old to have those fields, or does enterprise have a bug and isn't passing them through. If we see those fields as `nil`, we know something is wrong. This also applies to automated tests. If we forget to add fields to enterprise and hide them, our automated tests still pass so we don't realize we need to add something to enterprise.

Option 3: Missing fields are given a `nil` value.
Looks wrong because it is wrong, so a human will flag it when they see it. Once we add automated tests to enterprise, these tests will fail when suddenly there's a new field that enterprise doesn't set. Either way, hopefully we really would catch it during development time instead of RC or even release time.